soda_machine.rb

```diff
@@ -1,18 +1,46 @@
+# require ("soda.rb")
 class SodaMachine
   attr_reader :sodas, :cash
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
+    @sold_soda = nil
+    @brand = []
+    @index = 0
   end
 
```

Comment:
you don't need the `num_of_soda` variable here.  it's enough to just return `@sodas.count`